### PR TITLE
Migrate to operator-sdk v0.17.0

### PR DESCRIPTION
**Pull request by mrKisaoLamb**
- migrate to go.mod
- replace imports with correct one for new sdk
- define role permissions explicitly

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+	github.com/go-openapi/swag v0.19.6 // indirect
+	github.com/gogo/protobuf v1.3.1 // indirect
+	github.com/golang/groupcache v0.0.0-20200121045136-8c9f03a8e57e // indirect
+	github.com/google/gofuzz v1.1.0 // indirect
+	github.com/gophercloud/gophercloud v0.7.0 // indirect
+	github.com/hashicorp/golang-lru v0.5.4 // indirect
+	github.com/imdario/mergo v0.3.8 // indirect
+	github.com/json-iterator/go v1.1.9 // indirect
+	github.com/openstack-k8s-operators/nova-operator v0.0.0-20200416145254-c4da2ed23d04
+	github.com/operator-framework/operator-sdk v0.8.3-0.20190722210327-daf62d44e47e
```
Comment (bogdando) [inline go.mod:21]: Is this version correct? Shouldn't it be 0.17.0 or something?..

File: go.mod [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+	github.com/go-openapi/spec v0.19.5 // indirect
+	github.com/go-openapi/swag v0.19.6 // indirect
+	github.com/gogo/protobuf v1.3.1 // indirect
+	github.com/golang/groupcache v0.0.0-20200121045136-8c9f03a8e57e // indirect
+	github.com/google/gofuzz v1.1.0 // indirect
+	github.com/gophercloud/gophercloud v0.7.0 // indirect
+	github.com/hashicorp/golang-lru v0.5.4 // indirect
+	github.com/imdario/mergo v0.3.8 // indirect
+	github.com/json-iterator/go v1.1.9 // indirect
+	github.com/openstack-k8s-operators/nova-operator v0.0.0-20200416145254-c4da2ed23d04
```
Comment (bogdando) [inline go.mod:20]: I'm also not sure about this version of the operator itself...

File: go.mod (hunk 1 above)
Comment (mrKisaoLamb) [reply go.mod]: Done it was issue with generating from Gopkg

File: go.mod (hunk 2 above)
Comment (mrKisaoLamb) [reply go.mod]: done

**Comment by stuggi** [comment]
LGTM

**Pull request merged**